crates/turborepo-scm/src/package_deps.rs

```diff
@@ -20,7 +21,19 @@ impl SCM {
                 package_path,
                 inputs,
             ),
-            SCM::Git(git) => git.get_package_file_hashes(turbo_root, package_path, inputs),
+            SCM::Git(git) => git
+                .get_package_file_hashes(turbo_root, package_path, inputs)
+                .or_else(|e| {
+                    debug!(
+                        "failed to use git to hash files: {}. Falling back to manual",
+                        e
+                    );
+                    crate::manual::get_package_file_hashes_from_processing_gitignore(
```

Comment:
I carefully moved a lot of this fallback into "application" code inside of Go because doing the fallback lower-level caused the "we have three different ways that we hash files which are only mostly-compatible" situation.

I'd much prefer to hoist this back into application code out of `scm`.